proto/tendermint/abci/types.proto

```diff
@@ -338,6 +338,7 @@ message TxResult {
   uint32            index  = 2;
   bytes             tx     = 3;
   ResponseDeliverTx result = 4 [(gogoproto.nullable) = false];
+  bytes original_hash = 5;
```

Comment:
I modified abci.TxResult to include the original hash if the tx is malleated. This saves us from having to attempt to unwrap the malleated tx more than once. Alternatively, we could not cache the original hash, and unwrap the malleated transaction more than once.